src/domains/Lotto/index.js

```diff
@@ -1,43 +1,38 @@
-import {
-  MAX_LOTTO_NUMBER_COUNT,
-  MAX_LOTTO_NUMBER,
-  MIN_LOTTO_NUMBER,
-  ERROR_MESSAGES,
-} from "./constants/index.js";
+import { LottoNumber } from "../LottoNumber/index.js";
 
 export class Lotto {
-  #numbers;
+  static SIZE = 6;
+  static ErrorMessages = {
+    INVALID_LOTTO_NUMBERS: `로또는 ${Lotto.SIZE}개의 중복없는 숫자를 가진다`,
+  };
 
-  constructor(numbers) {
-    if (numbers.length !== MAX_LOTTO_NUMBER_COUNT) {
-      throw new Error(ERROR_MESSAGES.INVALID_COUNT);
-    }
-
-    if (new Set(numbers).size !== MAX_LOTTO_NUMBER_COUNT) {
-      throw new Error(ERROR_MESSAGES.DUPLICATE_NUMBERS);
-    }
+  #lottoNumbers;
 
-    if (
-      numbers.some(
-        (number) => number < MIN_LOTTO_NUMBER || number > MAX_LOTTO_NUMBER
-      )
-    ) {
-      throw new Error(ERROR_MESSAGES.OUT_OF_RANGE);
+  constructor(numbers) {
+    if (new Set(numbers).size !== Lotto.SIZE) {
+      throw new Error(Lotto.ErrorMessages.INVALID_LOTTO_NUMBERS);
     }
 
-    this.#numbers = numbers;
+    this.#lottoNumbers = new Set(
+      numbers.map((number) => new LottoNumber(number))
+    );
   }
 
   get numbers() {
-    return this.#numbers;
+    return Array.from(this.#lottoNumbers)
+      .map((lottoNumber) => lottoNumber.value)
+      .sort((a, b) => a - b);
+  }
+
+  countMatchingLottoNumber(lotto) {
+    return [...this.#lottoNumbers].filter((lottoNumber) =>
+      lotto.hasLottoNumber(lottoNumber)
+    ).length;
   }
 
-  evaluateLotto(winningNumbers, bonusNumber) {
-    return {
-      matchingCount: this.#numbers.filter((number) =>
-        winningNumbers.includes(number)
-      ).length,
-      isBonusNumberMatched: this.#numbers.includes(bonusNumber),
-    };
+  hasLottoNumber(lottoNumber) {
+    return [...this.#lottoNumbers].some((lottoNumberInLotto) =>
+      lottoNumberInLotto.isMatched(lottoNumber)
+    );
```

Comment:
지금 instance를 Set으로 관리하고 있기 때문에

```suggestion
  hasLottoNumber(lottoNumber) {
    return this.#lottoNumbers.has(new LottoNumber(lottoNumber))
```
이런식으로 표현해주면 어떨까요!?